diff --git a/dist/index.js b/dist/index.js
--- a/dist/index.js
+++ b/dist/index.js
@@ -13513,14 +13513,16 @@ var AbstractChat = class {
       this.setStatus({ status: "error", error: err });
     } finally {
       try {
-        (_b = this.onFinish) == null ? void 0 : _b.call(this, {
-          message: this.activeResponse.state.message,
-          messages: this.state.messages,
-          isAbort,
-          isDisconnect,
-          isError,
-          finishReason: (_a21 = this.activeResponse) == null ? void 0 : _a21.state.finishReason
-        });
+        if (this.activeResponse) {
+          (_b = this.onFinish) == null ? void 0 : _b.call(this, {
+            message: this.activeResponse.state.message,
+            messages: this.state.messages,
+            isAbort,
+            isDisconnect,
+            isError,
+            finishReason: (_a21 = this.activeResponse) == null ? void 0 : _a21.state.finishReason
+          });
+        }
       } catch (err) {
         console.error(err);
       }
diff --git a/dist/index.mjs b/dist/index.mjs
--- a/dist/index.mjs
+++ b/dist/index.mjs
@@ -13488,14 +13488,16 @@ var AbstractChat = class {
       this.setStatus({ status: "error", error: err });
     } finally {
       try {
-        (_b = this.onFinish) == null ? void 0 : _b.call(this, {
-          message: this.activeResponse.state.message,
-          messages: this.state.messages,
-          isAbort,
-          isDisconnect,
-          isError,
-          finishReason: (_a21 = this.activeResponse) == null ? void 0 : _a21.state.finishReason
-        });
+        if (this.activeResponse) {
+          (_b = this.onFinish) == null ? void 0 : _b.call(this, {
+            message: this.activeResponse.state.message,
+            messages: this.state.messages,
+            isAbort,
+            isDisconnect,
+            isError,
+            finishReason: (_a21 = this.activeResponse) == null ? void 0 : _a21.state.finishReason
+          });
+        }
       } catch (err) {
         console.error(err);
       }
diff --git a/src/ui/chat.ts b/src/ui/chat.ts
--- a/src/ui/chat.ts
+++ b/src/ui/chat.ts
@@ -757,14 +757,16 @@ export abstract class AbstractChat<UI_MESSAGE extends UIMessage> {
       this.setStatus({ status: 'error', error: err as Error });
     } finally {
       try {
-        this.onFinish?.({
-          message: this.activeResponse!.state.message,
-          messages: this.state.messages,
-          isAbort,
-          isDisconnect,
-          isError,
-          finishReason: this.activeResponse?.state.finishReason,
-        });
+        if (this.activeResponse) {
+          this.onFinish?.({
+            message: this.activeResponse.state.message,
+            messages: this.state.messages,
+            isAbort,
+            isDisconnect,
+            isError,
+            finishReason: this.activeResponse.state.finishReason,
+          });
+        }
       } catch (err) {
         console.error(err);
       }
